File: backend/src/main/java/com/jobpulse/service/ApplicationService.java
package com.jobpulse.service;

import com.jobpulse.dto.request.ApplicationRequest;
import com.jobpulse.dto.response.ApplicationResponse;
import com.jobpulse.dto.response.PagedResponse;
import com.jobpulse.dto.response.StatusHistoryResponse;
import com.jobpulse.entity.enums.ApplicationStatus;
import com.jobpulse.entity.enums.Priority;

import java.util.List;

public interface ApplicationService {
    ApplicationResponse createApplication(Long userId, ApplicationRequest request);
    List<ApplicationResponse> getAllApplications(Long userId);
    ApplicationResponse getApplicationById(Long userId, Long applicationId);
    ApplicationResponse updateApplication(Long userId, Long applicationId, ApplicationRequest request);
    void deleteApplication(Long userId, Long applicationId);
    ApplicationResponse updateStatus(Long userId, Long applicationId, ApplicationStatus newStatus);
    List<StatusHistoryResponse> getStatusHistory(Long userId, Long applicationId);
    PagedResponse<ApplicationResponse> getApplications(
            Long userId,
            String search,
            ApplicationStatus status,
            Priority priority,
            String location,
            int page,
            int size,
            String sortBy,
            String sortDir);
}

File: frontend/src/services/applicationService.js
import api from './api';

export const applicationService = {
  getApplications: async (params = {}) => {
    const response = await api.get('/applications', { params });
    return response.data;
  },
  getApplicationById: async (id) => {
    const response = await api.get(`/applications/${id}`);
    return response.data;
  },
  createApplication: async (data) => {
    const response = await api.post('/applications', data);
    return response.data;
  },
  updateApplication: async (id, data) => {
    const response = await api.put(`/applications/${id}`, data);
    return response.data;
  },
  updateStatus: async (id, status) => {
    const response = await api.patch(`/applications/${id}/status`, { status });
    return response.data;
  },
  deleteApplication: async (id) => {
    await api.delete(`/applications/${id}`);
  },
  getStatusHistory: async (id) => {
    const response = await api.get(`/applications/${id}/history`);
    return response.data;
  }
};
